### PR TITLE
Make /datum/hud garbage collect correctly

### PR DESCRIPTION
**Pull request by MrPerson**
For Dec. 21st:
The following path has failed to GC 739 time(s).
Path : /datum/hud

### CODE REVIEW

**Comment by duncathan** [comment]
Slowly but surely killing the ones that matter

**Comment by MrPerson** [comment]
You won't be so happy once github uncucks itself and shows you the diff

File: code/modules/mob/login.dm [hunk 1]
```diff
@@ -30,8 +30,9 @@
 	
 	client.images = null				//remove the images such as AIs being unable to see runes
 	client.screen = list()				//remove hud items just in case
-	if(hud_used)	qdel(hud_used)		//remove the hud objects
-	hud_used = new /datum/hud(src)
+	if(hud_used)
+		qdel(hud_used)					//remove the hud objects
```
Comment (Fox-McCloud) [inline code/modules/mob/login.dm]: shouldn't hud_used = null right after this, too?

Comment (duncathan) [inline code/modules/mob/login.dm]: The next line clears the reference by setting it to another instance.

Comment (Fox-McCloud) [inline code/modules/mob/login.dm]: fair enough; thanks.

**Pull request merged**